File: THCS/2022_2023/KhanhHoa_2223/Bai1/ATHORUA.cpp
#include <iostream>
#include <fstream>
#include <cmath>

using namespace std;

// Hàm tính số ngày leo để đạt đến đỉnh
int calculateDays(int V, int A, int B) {
    if (A >= V) return 1; // Nếu A >= V thì chỉ cần 1 ngày để leo đến đỉnh
    return ceil((double)(V - A) / (A - B)) + 1;
}

int main() {
    ifstream inputFile("ATHORUA.INP");
    ofstream outputFile("ATHORUA.OUT");

    int V, A1, B1, A2, B2;
    inputFile >> V >> A1 >> B1 >> A2 >> B2;

    // Tính số ngày cho Thỏ và Rùa
    int daysTho = calculateDays(V, A1, B1);
    int daysRua = calculateDays(V, A2, B2);

    // Xác định kết quả
    if (daysTho < daysRua) {
        outputFile << "THO";
    } else if (daysTho > daysRua) {
        outputFile << "RUA";
    } else {
        outputFile << "HOA";
    }

    inputFile.close();
    outputFile.close();

    return 0;
}

File: THCS/2022_2023/KhanhHoa_2223/Bai1/ATHORUA.py
import math

# Hàm tính số ngày leo để đạt đến đỉnh
def calculate_days(V, A, B):
    if A >= V:
        return 1  # Nếu A >= V thì chỉ cần 1 ngày để leo đến đỉnh
    return math.ceil((V - A) / (A - B)) + 1

# Đọc dữ liệu từ tệp ATHORUA.INP
with open("ATHORUA.INP", "r") as inputFile:
    V = int(inputFile.readline().strip())
    A1, B1 = map(int, inputFile.readline().strip().split())
    A2, B2 = map(int, inputFile.readline().strip().split())

# Tính số ngày cho Thỏ và Rùa
days_tho = calculate_days(V, A1, B1)
days_rua = calculate_days(V, A2, B2)

# Xác định kết quả
with open("ATHORUA.OUT", "w") as outputFile:
    if days_tho < days_rua:
        outputFile.write("THO")
    elif days_tho > days_rua:
        outputFile.write("RUA")
    else:
        outputFile.write("HOA")
